ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java

```diff
@@ -71,10 +71,15 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Map<Integer, List<ExprNodeGenericFuncDesc>> partitionSpecs = ParseUtils.getFullPartitionSpecs(root, table, conf, false);
     byte[] filterExp = null;
     if (partitionSpecs != null & !partitionSpecs.isEmpty()) {
-      // explicitly set expression proxy class to PartitionExpressionForMetastore since we intend to use the
+      // expression proxy class needs to be PartitionExpressionForMetastore since we intend to use the
       // filterPartitionsByExpr of PartitionExpressionForMetastore for partition pruning down the line.
-      conf.set(MetastoreConf.ConfVars.EXPRESSION_PROXY_CLASS.getVarname(),
-          PartitionExpressionForMetastore.class.getCanonicalName());
+      // Bail out early if expressionProxyClass is not configured properly.
+      String expressionProxyClass = conf.get(MetastoreConf.ConfVars.EXPRESSION_PROXY_CLASS.getVarname());
+      if (expressionProxyClass == null || expressionProxyClass.isEmpty() ||
+          !expressionProxyClass.equals(PartitionExpressionForMetastore.class.getCanonicalName())) {
```

Comment:
you could avoid the null/empty check by using: `!PartitionExpressionForMetastore.class.getCanonicalName()).equals(expressionProxyClass)`

Reply:
Fixed!